### DISCUSSION TITLE
How to set redis client resource in Spring Boot 2.5.9?

### DISCUSSION THREAD

**Discussion by devsungmin** [redis/lettuce#2411] [Q&A]
Hello.

I'm going to set up redis in springboot environment using lettuce.
We used the previous version of lettuce 5.3.7 to set up the project as follows.

I proceeded to set up the project by referring to the [official document of AWS.](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/BestPractices.Clients-lettuce.html)

``` kotlin
@Bean
    fun redisClientResources(): ClientResources {
        return DefaultClientResources.builder()
            .dnsResolver(DirContextDnsResolver())
            .reconnectDelay(
                Delay.fullJitter(
                    Duration.ofMillis(100),
                    Duration.ofSeconds(10),
                    100, TimeUnit.MILLISECONDS
                )
            )
            .build()
    }
```

However, in the process of setting up a new project, it was discovered that the version of lettuce was changed to 6.1.6 and that the `dnsResolver()` was Deprecated.
<img width="965" alt="스크린샷 2023-05-17 오후 5 38 28" src="https://github.com/lettuce-io/lettuce-core/assets/37958836/7ced6291-4472-4210-8ce0-e2eaafa47dad">

Could you tell me which method I should use to use the above settings in 6.1.6 version?

**Comment by mp911de** [comment]
The DNS resolver is deprecated because it resolves IP addresses synchronously. If you're good with JVM defaults regarding DNS resolution, there's no need to have a custom DNS resolver object. If you need customization, then you can either continue using Lettuce's DNS config mechanism or you can switch to netty's `AddressResolverGroup` for non-blocking DNS resolution.

**Comment by devsungmin** [comment]
Thank you @mp911de  😄
Your answer was very helpful for me to develop the application.
